rs/https_outcalls/adapter/src/rpc_server.rs

```diff
@@ -98,8 +114,148 @@ impl CanisterHttp {
         Self {
             client,
             socks_client,
+            cache: Arc::new(RwLock::new(BTreeMap::new())),
             logger,
             metrics: AdapterMetrics::new(metrics),
+            http_connect_timeout_secs: config.http_connect_timeout_secs,
+        }
+    }
+
+    fn create_socks_client_for_address(
```

Comment:
Since this is a member function I would slightly change it. I would call it

maybe_add_socks_client_for_addr(...) change the implementation. I believe the code will be less since you won't need to handle the errors.

Reply:
Maybe I missed the point. It being a member function is not of much help, as we need to hold the lock. I've restructured the code a bit .